pkg/config/config.go

```diff
@@ -1334,6 +1334,12 @@ func (c *Config) FindHelperBinary(name string, searchPATH bool) (string, error)
 		if fi, err := os.Stat(fullpath); err == nil && fi.Mode().IsRegular() {
 			return fullpath, nil
 		}
+		abspath, err := filepath.Abs(fullpath)
+		if err == nil {
+			if lp, err := exec.LookPath(abspath); err == nil {
+				return lp, nil
+			}
+		}
```

Comment:
Thinking about this. If we have absolute path then use builtin machinery, but as a fallback use old implementation.
```suggestion
		if filepath.IsAbs(fullpath) {
			if lp, err := exec.LookPath(fullpath); err == nil {
				return lp, nil
			}
		} else {
			if fi, err := os.Stat(fullpath); err == nil && fi.Mode().IsRegular() {
				return fullpath, nil
			}
		}
```

Reply:
I don't understand either solution, this function by definition should not lookup in $PATH unless `searchPATH` was set to true.

Reply:
From experiments, LookPath doesn't traverse the PATH records if absolute path is given as input. For Windows OS it still extends the search from `C:\something\executablename` to `C:\something\executablename.exe`, `C:\something\executablename.bat`, etc. And then return the updated version (with the extension) as an output, if one is found.

Reply:
I can put it behind `if runtime.GOOS = "windows"` as it seems this doesn't really change the behavior of the current implementation for non-Windows OSes.

Reply:
The end goal is to make it understand that "C:\Program Files\Red Hat\Podman\gvproxy" should resolve to "C:\Program Files\Red Hat\Podman\gvproxy.exe". So, that it will be possible to use the same binary name `gvproxy` (which is now hardcoded, and it looks fine to have it this way) w/o any conditional code in Podman, because this method will behave more like a LookPath, which is capable of understanding that for example `ssh` is actually `ssh.exe` on Windows.

Reply:
I understand the goal, but wouldn't it like this just include an extra unnecessary stat() call on unix.
The simple solution seems to just append `.exe` when on windows. At least this is something I could understand without a comment.

The current code would need a big comment explaining what it does.

Reply:
If this is a big no-go for the library, then it is also fine. All such corner cases could have a 3 lines workaround in the caller and we can just pass on this PR until more demand of this change.

Reply:
It is good to fix it here. All I am saying is it is very not obvious how this code works. If we only pass full path to LookPath it should keep the promise of not looking in $PATH I assume?

But can't you just remove
```
fullpath := filepath.Join(path, name)
if fi, err := os.Stat(fullpath); err == nil && fi.Mode().IsRegular() {
	return fullpath, nil
}
```

with this new code, passing the full path to LookupPath() should also work on linux/unix?

Reply:
> But can't you just remove

If we filter directories, which are not given as absolute paths or use filepath.Abs and ignore error cases (I don't know if here we could satisfy the conditions for this to fail, but we have to take into account the error case). Because now it gives potential escape via relative path from the PWD, but if we pass relative ones into LookPath it is a wired range of path escapes. And not-validated for being absolute path CONTAINERS_HELPER_BINARY_DIR is the first candidate to be a trouble maker:
```
	if dir, found := os.LookupEnv("CONTAINERS_HELPER_BINARY_DIR"); found {
		dirList = append([]string{dir}, dirList...)
	}
```

Not related to this PR, but it now looks to me that CONTAINERS_HELPER_BINARY_DIR deserves some sanity checks.

Reply:
> passing the full path to LookupPath() should also work on linux/unix?

Yes, that seems to be a good approach: Always do `filepath.Abs(filepath.Join(…))` + `exec.LookPath`. There would be not much extra cost on Unix (`filepath.Abs` is just `os.Getwd` + some in-memory manipulation).

The returned path from this function could now be absolute in most cases, but that’s hardly a bad thing.

---

Either way, the `LookupEnv` call does need a quite verbose comment explaining that this is important to preserve, because on Windows it turns a command name into an executable path (handling `$PATHEXT`) — perhaps even with an example.

Otherwise someone like me would be pretty likely to remove it in a few months.

Reply:
> I will revisit the changes this weekend and will do extended testing on non-Windows hosts.

Having a unit test (at least a single case combining a path + a fictitious executable `t.Tempdir()`) would be nice as well. OTOH the test would have to be, annoyingly, platform-specific.

(I have no opinion on whether the unit test should be a required part of this PR.)

Reply:
Simplified implementation (always use Abs + LookPath) and added comments.

Proofs that LookPath always short circuits for absolute paths:
* Windows https://github.com/golang/go/blob/213495a4a67c318a1fab6e76093e6690c2141c0e/src/os/exec/lp_windows.go#L82
* Unix https://github.com/golang/go/blob/213495a4a67c318a1fab6e76093e6690c2141c0e/src/os/exec/lp_unix.go#L57

This implementation is slightly superior to original one on Unix in the form that it taxed into account executable bit and on Windows it correctly loops through PATHEXT, when needed.

I'm not sure if this needs unit test then, because golang repo looks like a more appropriate way to place them (we want to validate platform behavior).